### PR TITLE
Added support for MSVC

### PR DESCRIPTION
**Pull request by marler8997**
Some ideas on how to support MSVC.  Feel free to pick this apart and indicate which parts you would or would not want to merge.

### CODE REVIEW

**Comment by thejoshwolfe** [comment]
Specifically, you mean building the Zig compiler with MSVC, right? We're not talking about using the Zig compiler to build user programs that were made for an MSVC build environment or something.

**Comment by marler8997** [comment]
Correct, building zig using MSVC

**Comment by andrewrk** [comment]
Thanks for the patch. I'm installing windows on a laptop I have lying around and I'm going to try this out. We do need to get quadmath working to get the tests passing.

**Comment by marler8997** [comment]
Yeah, and I still haven't finished the build, it seems command line to finally link zig.exe it too long.  Apparently this has been a problem for other projects, there's supposed to be a way to enable putting the command line arguments in a file but I'm still working on it.

**Comment by andrewrk** [comment]
I've got LLVM building in MSVC right now. Next is clang, then I'll try your zig changes.

> Specifically, you mean building the Zig compiler with MSVC, right? We're not talking about using the Zig compiler to build user programs that were made for an MSVC build environment or something.

To be clear we do support this use case. `--target-environ msvc` makes "C ABI" mean "MSVC ABI".

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,13 +26,86 @@ option(ZIG_TEST_COVERAGE "Build Zig with test coverage instrumentation" OFF)
...
 option(ZIG_FORCE_EXTERNAL_LLD "If your system has the LLD patches use it instead of the embedded LLD" OFF)
 
+find_package(LLVM REQUIRED CONFIG)
+include_directories(${LLVM_INCLUDE_DIRS})
 
-find_package(llvm)
-include_directories(${LLVM_INCLUDE_DIR})
-link_directories(${LLVM_LIBDIRS})
+# TODO: this currently doesn't work, it currently defines UNICODE but zig
+#       uses MBCS
```
Comment (andrewrk) [inline CMakeLists.txt]: I read the docs about MBCS but it's not quite clear to me. What is zig doing to choose MBCS? Would it be better to choose UNICODE?

**Pull request merged**